engine/src/main/java/org/terasology/rendering/logic/MeshRenderer.java

```diff
@@ -218,17 +223,17 @@ private void renderEntitiesByMaterial(SetMultimap<Material, EntityRef> meshByMat
                             lastMesh.preRender();
                         }
 
-                        Matrix4f modelViewMatrix = MatrixUtils.calcModelViewMatrix(JomlUtil.from(worldRenderer.getActiveCamera().getViewMatrix()), matrixCameraSpace);
-                        MatrixUtils.matrixToFloatBuffer(modelViewMatrix, tempMatrixBuffer44);
-                        MatrixUtils.matrixToFloatBuffer(MatrixUtils.calcNormalMatrix(modelViewMatrix), tempMatrixBuffer33);
+                        modelViewMatrix.set(worldRenderer.getActiveCamera().getViewMatrix()).transpose().mul(matrixCameraSpace);
```

Comment:
Why is there a `transpose` inserted here? This seems to be actually changing the logic.

Reply:
the transpose is not needed anymore when I pull in the camera from upstream.